pkg/lib/_internal/packagekit.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { InstallProgressCB, MissingPackages, PackageManager, ProgressCB } from './packagemanager-abstract';
+import * as PK from "packagekit.js";
+
+export class PackageKitManager implements PackageManager {
```

Comment:
What a nice and clean implementation! Shows that the abstract API is very much tailored towards PK, but that's ok.